src/gen_rules.ml

```diff
@@ -225,9 +225,15 @@ module Gen(P : sig val sctx : Super_context.t end) = struct
         List.iter js_targets ~f:(fun js_target ->
           assert (Path.Build.equal (Path.Build.parent_exn js_target)
                     ctx_dir));
-        Predicate.create ~id ~f:(fun basename ->
-          not (List.exists js_targets ~f:(fun js_target ->
-            String.equal (Path.Build.basename js_target) basename)))
+        let f =
```

Comment:
Let's leave a comment here to remove this hack for 2.0. It should no longer be necessary for targets to be filtered out manually.

Reply:
Done.